File: src/01-easy/ascii-art/solution.dart
import 'dart:io';
void main(){
  int L=int.parse(stdin.readLineSync()!), H=int.parse(stdin.readLineSync()!);
  String T=stdin.readLineSync()!.toUpperCase();
  List<String> rows=List.generate(H,(_)=>stdin.readLineSync()!);
  for(int i=0;i<H;i++){
    StringBuffer sb=StringBuffer();
    for(int j=0;j<T.length;j++){
      int idx=T.codeUnitAt(j)-65;if(idx<0||idx>25)idx=26;
      sb.write(rows[i].substring(idx*L,idx*L+L));
    }
    print(sb);
  }
}

File: src/01-easy/ascii-art/solution.rs
use std::io::{self,BufRead};
fn main(){
    let stdin=io::stdin();let mut lines=stdin.lock().lines();
    let l:usize=lines.next().unwrap().unwrap().trim().parse().unwrap();
    let h:usize=lines.next().unwrap().unwrap().trim().parse().unwrap();
    let t:Vec<usize>=lines.next().unwrap().unwrap().to_uppercase().bytes()
        .map(|b|{let i=(b as i32)-65;if i<0||i>25{26}else{i as usize}}).collect();
    let rows:Vec<String>=(0..h).map(|_|lines.next().unwrap().unwrap()).collect();
    for row in &rows{
        let mut out=String::new();
        for &idx in &t{out.push_str(&row[idx*l..idx*l+l]);}
        println!("{}",out);
    }
}
